src/inference/src/compilation_context.cpp

```diff
@@ -121,8 +122,32 @@ std::string ModelCache::compute_hash(const std::string& modelStr,
 
         auto ptr = static_cast<size_t*>(tensor.data());
         size_t size = tensor.get_size() / sizeof(size_t);
-        for (size_t i = 0; i < size; i++)
-            seed = hash_combine(seed, ptr[i]);
+
+        size_t block_size = 100000000;
```

Comment:
100000000 is optimum for 16GB weights file. In this case we have blocks_num = 22
This value is discussible.

Reply:
what if we have 1GB model? does it make sense to perform parallal_for across all threads?
If yes, block_size should be smaller. 

maybe we need to divide total_size by number of threads, but still keep minimal block_size as certain value? In this case, we split work across multiple threads, but if work is small enough, only a few threads will be working.

Reply:
Updated block_size to make it 1MB or 1000000 / sizeof(size_t)